src/App.tsx

```diff
@@ -1,24 +1,54 @@
 import React from 'react';
+import { Clock } from './components/Clock/Clock';
+
 import './App.scss';
 
-const App: React.FC = () => {
-  const timerId: NodeJS.Timer = setInterval(() => {
-    const date: Date = new Date();
-
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
-};
+class App extends React.Component {
+  state = {
+    isClockVisible: true,
+    clockName: Math.round(Math.random() * 1000),
+  };
+
+  showClock = () => {
+    this.setState({
+      isClockVisible: true,
+    });
+  };
+
+  hideClock = () => {
+    this.setState({
+      isClockVisible: false,
+    });
+  };
+
+  renameClock = () => {
+    this.setState({
+      clockName: Math.round(Math.random() * 1000),
+    });
+  };
+
+  render() {
+    const { clockName, isClockVisible } = this.state;
+
+    return (
+      <div className="App">
+        <h1>React clock</h1>
+        {isClockVisible && <Clock clockName={clockName} />}
+
+        <button type="button" onClick={this.showClock} disabled={isClockVisible}>
+          Show Clock
+        </button>
+
+        <button type="button" onClick={this.hideClock} disabled={!isClockVisible}>
+          Hide Clock
+        </button>
```

Comment:
Remember bout proper formatting, if you have to many attributes - put them on a newline
```suggestion
        <button 
        	type="button"
        	onClick={this.showClock}
        	disabled={isClockVisible}
        >
          Show Clock
        </button>

        <button 
        	type="button"
        	onClick={this.hideClock} 
        	disabled={!isClockVisible}
        >
          Hide Clock
        </button>
```